### PR TITLE
update azdata-test in notebooks extension

### PR DESCRIPTION
**Pull request by kisantia**
This updates the azdata-test dependency in the notebooks extension to fix this dependabot alert: https://github.com/microsoft/azuredatastudio/security/dependabot/603

There were a few updates that needed to be made since a few compilation errors started after the azdata-test package was bumped, since it also bumped several other dependencies. There were the errors: 
image

The fix was to cast the result from those `yaml.safeLoad()` calls. It looks like the errors started showing up because the resolved version of  @types/js-yaml went from 3.12.1 to 3.12.10 and the return type changed from `any` to `string | object | undefined`

3.12.1:
image
3.12.10
image

I also ran the notebook extension unit tests locally and confirmed that they're still passing. The extension unit tests aren't currently running in the pipelines, so that's what I'll look into enabling again next...

### CODE REVIEW

File: extensions/notebook/src/book/bookTocManager.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class BookTocManager implements IBookTocManager {
 		}
 
 		for (const [key, value] of this.tocFiles.entries()) {
-			const yamlFile = await yaml.safeLoad(value);
```
Comment (kisantia) [inline extensions/notebook/src/book/bookTocManager.ts:153]: this wasn't related to the package version update, but it was a red squiggly when I was in the file, so updated this while I was there

File: extensions/notebook/src/book/bookModel.ts [hunk 2]
```diff
@@ -183,7 +183,7 @@ export class BookModel {
 		if (this._tableOfContentsPath) {
 			try {
 				let fileContents = await fsPromises.readFile(this._configPath, 'utf-8');
-				const config = yaml.safeLoad(fileContents.toString());
+				const config = yaml.safeLoad(fileContents.toString()) as JupyterBookSection;
```
Comment (kisantia) [inline extensions/notebook/src/book/bookModel.ts:186]: @barbaravaldez could you double check if this cast and the one below in `bookTocManager.ts` are correct or should be changed? The variables in the tests have types specified, but wasn't sure about these

**Review by cheenamalhotra — APPROVED** [review]
It should be safe IMO, but it looks like yarn.lock has a lot of package updates arriving with this PR..

**Review by corivera — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

File: extensions/notebook/src/book/bookModel.ts (hunk 2 above)
Comment (corivera) [reply extensions/notebook/src/book/bookModel.ts:186]: JupyterBookSection is the only class I'm seeing with a title field for this one, and the tableOfContents variable in bookTocManager already had a JupyterBookSection[] type. Looks good to me.

**Review by barbaravaldez — APPROVED** [review]

File: extensions/notebook/src/book/bookModel.ts (hunk 2 above)
Comment (barbaravaldez) [reply extensions/notebook/src/book/bookModel.ts:186]: Yup! It looks good to me

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**